src/s3contents.ts

```diff
@@ -1114,11 +1087,17 @@ export class Drive implements Contents.IDrive {
         );
         path = isDir ? path + '/' : path;
 
-        Contents.forEach(c => {
+        const promises = Contents.map(async c => {
           const remainingFilePath = c.Key!.substring(path.length);
           // copy each file from old directory to new location
-          this.copy_file(remainingFilePath, path, newFileName, bucketName);
+          await this.copy_file(
```

Comment:
Here you need to `return` the `copy_file` promise instead of `await`ing it.
```suggestion
          return this.copy_file(
```